app/bootstrap/internal/state/state.go

```diff
@@ -0,0 +1,107 @@
+//    \\ SPIKE: Secure your secrets with SPIFFE. — https://spike.ist/
+//  \\\\\ Copyright 2024-present SPIKE contributors.
+// \\\\\\\ SPDX-License-Identifier: Apache-2.0
+
+package state
+
+import (
+	"crypto/rand"
+	"os"
+	"strconv"
+
+	"github.com/cloudflare/circl/group"
+	shamir "github.com/cloudflare/circl/secretsharing"
+	"github.com/spiffe/spike-sdk-go/crypto"
+	"github.com/spiffe/spike-sdk-go/log"
+
+	"github.com/spiffe/spike/app/bootstrap/internal/env"
+	"github.com/spiffe/spike/app/bootstrap/internal/validation"
+)
+
+// RootShares generates a set of Shamir secret shares from a cryptographically
+// secure random root key. It creates a 32-byte random seed, uses it to generate
+// a root secret on the P256 elliptic curve group, and splits it into n shares
+// using Shamir's Secret Sharing scheme with threshold t. The threshold t is
+// set to (ShamirThreshold - 1), meaning t+1 shares are required for
+// reconstruction. A deterministic reader seeded with the root key is used to
+// ensure identical share generation across restarts, which is critical for
+// synchronization after crashes. The function performs security validation and
+// zeroing of sensitive data after use.
+func RootShares() []shamir.Share {
+	const fName = "rootShares"
+
+	var rootKeySeed [32]byte
+	if _, err := rand.Read(rootKeySeed[:]); err != nil {
+		log.Fatal(err.Error())
+	}
+
+	// Initialize parameters
+	g := group.P256
+	t := uint(env.ShamirThreshold() - 1) // Need t+1 shares to reconstruct
+	n := uint(env.ShamirShares())        // Total number of shares
+
+	log.Log().Info(fName, "t", t, "n", n)
+
+	// Create a secret from our 32-byte key:
+	rootSecret := g.NewScalar()
+
+	if err := rootSecret.UnmarshalBinary(rootKeySeed[:]); err != nil {
+		log.FatalLn(fName + ": Failed to unmarshal key: %v" + err.Error())
+		os.Exit(1)
+	}
```

Comment:
There might be some code duplication since I copied these functions from Nexus into the `internal` modules of this app. -- Once we are sure things work, we can do a follow-up clean-up to consolidate common code and remove the code that is not used anymore.